src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -1820,6 +1907,253 @@ public string DeleteCounter(DocumentsOperationContext context, string documentId
+                {
+                    if (tx.LowLevelTransaction.Committed == false)
+                        return;
+
+                    // not sure if we can _rely_ on the tx write lock here, so let's be safe and create
+                    // a new instance, just in case
+                    _tableCreated = new HashSet<string>(_tableCreated, StringComparer.OrdinalIgnoreCase)
+                    {
+                        tableName
+                    };
+                };
+            }
+
+            return tx.OpenTable(tableSchema, tableName);
+        }
+
+        private void CreateCounterTombstone(DocumentsOperationContext context, Slice documentKeyPrefix, Slice counterNameSlice, CollectionName collectionName, string remoteChangeVector = null)
+        {
+            var table = GetOrCreateCounterTombstonesTable(context.Transaction.InnerTransaction, collectionName);
+
+            var etag = _documentsStorage.GenerateNextEtag();
+            var changeVector = remoteChangeVector ?? _documentsStorage.GetNewChangeVector(context, etag);
+
+            using (context.Allocator.Allocate(documentKeyPrefix.Size + counterNameSlice.Size, out var counterKeyBuffer))
+            using (CreateCounterKeySlice(context, counterKeyBuffer, documentKeyPrefix, counterNameSlice, out var counterTombstoneKeySlice))
+            {
+                if (table.ReadByKey(counterTombstoneKeySlice, out var tableValueReader))
+                {
+                    var existingChangeVector = ExtractCounterTombstoneChangeVector(context, ref tableValueReader);
+                    if (ChangeVectorUtils.GetConflictStatus(changeVector, existingChangeVector) == ConflictStatus.AlreadyMerged)
+                    {
+                        // do nothing...
+                        return;
+                    }
+                }
+
+                using (table.Allocate(out TableValueBuilder tvb))
+                using (Slice.From(context.Allocator, changeVector, out var cv))
+                using (DocumentIdWorker.GetStringPreserveCase(context, collectionName.Name, out Slice collectionSlice))
+                {
+                    tvb.Add(counterTombstoneKeySlice);
+                    tvb.Add(Bits.SwapBytes(etag));
+                    tvb.Add(cv);
+                    tvb.Add(collectionSlice);
+                    tvb.Add(context.GetTransactionMarker());
+
+                    table.Set(tvb);
+                }
+            }
+        }
+
+        private static string ExtractCounterTombstoneChangeVector(DocumentsOperationContext context, ref TableValueReader reader)
+        {
+            var changeVectorPtr = reader.Read((int)CounterTombstonesTable.ChangeVector, out int changeVectorSize);
+            return Encoding.UTF8.GetString(changeVectorPtr, changeVectorSize);
+        }
+
+        public IEnumerable<CounterTombstoneDetail> GetCounterTombstonesForDoc(DocumentsOperationContext context, string docId)
```

Comment:
why do you need this? I see it is used only in tests and also there you need that only for the count/exists?
if you need that only for the count/exists, please simplify this

Reply:
done (removed)